From 5e64600811935e7f07a2cbba8a0bc1392c673439 Mon Sep 17 00:00:00 2001
From: Alwin Esch <alwin.esch@web.de>
Date: Thu, 26 Mar 2020 22:05:36 +0100
Subject: [PATCH] [addons] remove deprecated xbmc_addon_types.h

This has been done at AddonBase.h for a long time. Now that all addons have removed this, it can also be removed on Kodi.

There are no changes to the API, only it had to be corrected in three old headers.
---
 .../kodi-addon-dev-kit/include/kodi/CMakeLists.txt    |  1 -
 .../kodi-addon-dev-kit/include/kodi/kodi_vfs_types.h  |  2 +-
 .../addons/kodi-addon-dev-kit/include/kodi/versions.h |  1 -
 .../include/kodi/xbmc_addon_types.h                   | 11 -----------
 .../kodi-addon-dev-kit/include/kodi/xbmc_pvr_types.h  |  2 +-
 5 files changed, 2 insertions(+), 15 deletions(-)
 delete mode 100644 xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_addon_types.h

diff --git a/xbmc/addons/kodi-addon-dev-kit/include/kodi/CMakeLists.txt b/xbmc/addons/kodi-addon-dev-kit/include/kodi/CMakeLists.txt
index 07bdc66b4722..47d54d70568f 100644
--- a/xbmc/addons/kodi-addon-dev-kit/include/kodi/CMakeLists.txt
+++ b/xbmc/addons/kodi-addon-dev-kit/include/kodi/CMakeLists.txt
@@ -11,7 +11,6 @@ set(HEADERS AddonBase.h
             libXBMC_pvr.h
             versions.h
             xbmc_addon_dll.h
-            xbmc_addon_types.h
             xbmc_epg_types.h
             xbmc_pvr_dll.h
             xbmc_pvr_types.h)
diff --git a/xbmc/addons/kodi-addon-dev-kit/include/kodi/kodi_vfs_types.h b/xbmc/addons/kodi-addon-dev-kit/include/kodi/kodi_vfs_types.h
index c1843b0bd448..8bc5873863be 100644
--- a/xbmc/addons/kodi-addon-dev-kit/include/kodi/kodi_vfs_types.h
+++ b/xbmc/addons/kodi-addon-dev-kit/include/kodi/kodi_vfs_types.h
@@ -17,7 +17,7 @@
 #endif
 
 #include <stdint.h>
-#include "xbmc_addon_types.h"
+#include "AddonBase.h"
 #ifdef BUILD_KODI_ADDON
 #include "IFileTypes.h"
 #else
diff --git a/xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h b/xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h
index 2ecc14d092ae..ec49ed114776 100644
--- a/xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h
+++ b/xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h
@@ -35,7 +35,6 @@
 #define ADDON_GLOBAL_VERSION_MAIN_XML_ID              "kodi.binary.global.main"
 #define ADDON_GLOBAL_VERSION_MAIN_DEPENDS             "AddonBase.h" \
                                                       "xbmc_addon_dll.h" \
-                                                      "xbmc_addon_types.h" \
                                                       "libXBMC_addon.h" \
                                                       "addon-instance/"
 
diff --git a/xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_addon_types.h b/xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_addon_types.h
deleted file mode 100644
index bfd85c25b8c1..000000000000
--- a/xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_addon_types.h
+++ /dev/null
@@ -1,11 +0,0 @@
-/*
- *  Copyright (C) 2005-2018 Team Kodi
- *  This file is part of Kodi - https://kodi.tv
- *
- *  SPDX-License-Identifier: GPL-2.0-or-later
- *  See LICENSES/README.md for more information.
- */
-
-#pragma once
-
-#include "AddonBase.h" // compatibility fallback
diff --git a/xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_types.h b/xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_types.h
index 0c16e07fc828..f4d7383fa4e3 100644
--- a/xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_types.h
+++ b/xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_types.h
@@ -20,7 +20,7 @@
 #include <stdint.h>
 #include <stdio.h>
 
-#include "xbmc_addon_types.h"
+#include "AddonBase.h"
 #include "xbmc_epg_types.h"
 
 /*! @note Define "USE_DEMUX" at compile time if demuxing in the PVR add-on is used.
